graph-macro/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+use proc_macro::TokenStream;
+use quote::{format_ident, quote};
+use syn;
+
+#[proc_macro_attribute]
+pub fn odata_next_link(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let input = syn::parse_macro_input!(item as syn::ItemStruct);
+    let name = &input.ident;
+
+    let plural = format_ident!("{}s", name);
+
+    let result = quote! {
+        #input
+
+        impl ODataNextLink<#name> for #name {
```

Comment:
I wonder if this might be better as a derive macro.

Reply:
Semanticaly, derive macro are expected to only implement a trait for the type, even if it's not a hard rule.


At first I went for the proc_macro_attribute instead of the derive macro because derive macro  can't modify the item but in the end I didn't need that capability. I can try to change that to a derive macro if you want.

Reply:
I guess I was thinking more derive macro because you could set attributes for a next link field given any struct. 

```rust
#[derive(ODataLink)]
pub struct Users {
  #[ODataLink(odata_next_link)]
  pub next_link: Option<String>
  #[ODataLink(odata_value)]
  pub value: Option<string>
}
```

And then use those fields.

Reply:
We could provide a derive macro for users who want to declare their struct `Users` instead using one provided by the odata_next_link proc macro (on the `User` struct) , but it won't replace it.

I started working on it a bit and it's not trivial at all : we need to find the correct attribute for value (either by name or with the custom macro attribute) and find the type inside the Vec. But we need to handle every cases : `Vec<User>`,  `std::vec::Vec<User>`, `Vec<User<T>>` , etc.